src/nms-api/ISession.cs

```diff
@@ -16,6 +16,7 @@
  */
 
 using System;
+using System.Threading.Tasks;
```

Comment:
What about CreateMessage do we want async varients on these? I know for AMQP its probably super fast to return as its all done within the process, but in some other protocols hilariously to create a message a remote call is made (yeah! its shocking but the old protocols are old for a reason!)